apps/site/pages/ltb/entry/[slug]/index.tsx

```diff
@@ -0,0 +1,1041 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { getSession } from "next-auth/react";
+import { GetServerSideProps } from "next";
+import { PencilIcon, StarIcon, TrashIcon } from "@heroicons/react/24/solid";
+import { Dialog, showToast } from "@self-learning/ui/common";
+import Image from "next/image";
+import { trpc } from "@self-learning/api-client";
+import { formatTimeIntervalToString } from "@self-learning/util/common";
+import { MarkdownEditorDialog, MarkdownViewer } from "@self-learning/ui/forms";
+import { useForm, Controller, FormProvider, useFormContext } from "react-hook-form";
+import {
+	getLearningDiaryInformation,
+	LearningDiaryEntryResult,
+	LearningDiaryInformation,
+	LearningStrategy,
+	LearningTechnique,
+	LearningTechniqueEvaluation
+} from "@self-learning/api";
+import { useRouter } from "next/router";
+
+export default function LearningDiaryEntryOverview({
+	learningDiaryInformation,
+	slug
+}: {
+	learningDiaryInformation: LearningDiaryInformation;
+	slug: string | null;
+}) {
+	const router = useRouter();
+
+	const [currentIndex, setCurrentIndex] = useState<number>(() => {
+		const initialIndex =
+			slug && !isNaN(Number(slug))
+				? Number(slug) - 1
+				: learningDiaryInformation?.learningDiaryEntries?.length
+				? learningDiaryInformation.learningDiaryEntries.length - 1
+				: 0;
+		return initialIndex < 0 ? 0 : initialIndex;
+	});
+
+	const [entries, setEntries] = useState<LearningDiaryEntryResult[]>(
+		learningDiaryInformation.learningDiaryEntries
+	);
+
+	const handleUpdateEntry = (updatedEntry: LearningDiaryEntryResult) => {
+		setEntries(prevEntries =>
+			prevEntries.map(entry => (entry.id === updatedEntry.id ? updatedEntry : entry))
+		);
+	};
+
+	useEffect(() => {
+		window.history.pushState({}, "", "/ltb/entry/" + (currentIndex + 1));
+	}, [currentIndex]);
+
+	return (
+		<div className="flex justify-center">
+			<div className="w-2/3 py-4">
+				<div className="mb-4 flex justify-center">
+					<EntrySwitcher
+						maxLength={entries.length}
+						setIndex={setCurrentIndex}
+						currentIndex={currentIndex}
+					/>
+				</div>
+				<LearningDiaryEntryForm
+					entry={entries[currentIndex]}
+					learningDiaryInformation={learningDiaryInformation}
+					onUpdate={handleUpdateEntry}
+				/>
+			</div>
+		</div>
+	);
+}
+
+function EntrySwitcher({
+	maxLength,
+	setIndex,
+	currentIndex
+}: {
+	maxLength: number;
+	setIndex: (index: number) => void;
+	currentIndex: number;
+}) {
+	const handlePrev = () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-expect-error
+		setIndex(prevIndex => Math.max(prevIndex - 1, 0));
+	};
+
+	const handleNext = () => {
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-expect-error
+		setIndex(prevIndex => Math.min(prevIndex + 1, maxLength - 1));
+	};
+
+	return (
+		<div className="flex space-x-16">
```

Comment:
das styling (space-x-16) mit in die obere komponenten ziehen, macht sie mehr reusable